ic-stable-structures/src/memory_mapped_files/memory.rs

```diff
@@ -12,56 +15,87 @@ const WASM_PAGE_SIZE_IN_BYTES: u64 = 65536;
 pub struct MemoryMappedFileMemoryManager {
     base_path: PathBuf,
     is_persistent: bool,
+    created_memory_resources: RwLock<BTreeMap<PathBuf, MemoryMappedFileMemory>>,
 }
 
 impl MemoryMappedFileMemoryManager {
     pub fn new(base_path: PathBuf, is_persistent: bool) -> Self {
         Self {
             base_path,
             is_persistent,
+            created_memory_resources: Default::default(),
+        }
+    }
+
+    pub fn flush_and_save_copies_to(&self, path: impl AsRef<Path>) -> Result<(), MemMapError> {
+        let created_memory_resources = self.created_memory_resources.read();
+        for (file_path, memory) in created_memory_resources.iter() {
+            let file_name = file_path
```

Comment:
maybe here you can direectly convert file_name to str

Reply:
I need to extract the name of the file first, so it's more convenient to do it first and then convert it to string